File: src/main/java/cube8540/oauth/authentication/credentials/oauth/token/application/DefaultAccessTokenReadService.java
package cube8540.oauth.authentication.credentials.oauth.token.application;

import cube8540.oauth.authentication.credentials.oauth.token.domain.read.Oauth2AccessTokenReadRepository;
import cube8540.oauth.authentication.credentials.oauth.token.domain.read.model.AccessTokenDetailsWithClient;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class DefaultAccessTokenReadService implements AccessTokenReadService {

    private final Oauth2AccessTokenReadRepository repository;

    @Autowired
    public DefaultAccessTokenReadService(Oauth2AccessTokenReadRepository repository) {
        this.repository = repository;
    }

    @Override
    public List<AccessTokenDetailsWithClient> getAuthorizeAccessTokens(Authentication authentication) {
        return repository.readAccessTokenWithClientByUsername(authentication.getName());
    }
}

File: src/main/kotlin/cube8540/oauth/authentication/credentials/oauth/token/application/DefaultAccessTokenReadService.kt
package cube8540.oauth.authentication.credentials.oauth.token.application

import cube8540.oauth.authentication.credentials.oauth.token.domain.AccessTokenDetailsWithClient
import cube8540.oauth.authentication.credentials.oauth.token.domain.OAuth2AccessTokenReadRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class DefaultAccessTokenReadService @Autowired constructor(
    private val repository: OAuth2AccessTokenReadRepository
): AccessTokenReadService {

    @Transactional(readOnly = true)
    override fun getAuthorizeAccessTokens(username: String): List<AccessTokenDetailsWithClient> =
        repository.readAccessTokenWithClientByUsername(username)
}